scripts/jinja_gen.py

```diff
@@ -31,10 +31,21 @@ def get_file_contents(filepath):
     parser.add_argument('--gst_udp_port', default=5600, help="Gstreamer UDP port for SITL")
     parser.add_argument('--video_uri', default=5600, help="Mavlink camera URI for SITL")
     parser.add_argument('--mavlink_cam_udp_port', default=14530, help="Mavlink camera UDP port for SITL")
+    parser.add_argument('--ros2-distro', default='', dest='ros2_distro', type=str,
+                    help="ROS2 distro, only required if generating the agent for usage with ROS2 nodes, by default empty")
```

Comment:
Where is this being passed?

Reply:
The 'ros2_distro' argument is additional feature in current version.
Someone may enter the value for specific reason.
If no value is entered, the jinja_gen.py find ros version in line 40 - 47 using environment variable 'ROS_DISTRO'.

Actually, I added the argument to parser because the line 40 - 47 (from [generate_microRTPS_bridge.py](https://github.com/PX4/PX4-Autopilot/blob/master/msg/tools/generate_microRTPS_bridge.py)) code used argument parser.

I think the add_argument for 'ros2_distro' is not mandatory codes if line 45 ('ros2_distro = args.ros2_distro') is removed.

We can make decision about this. There is two options.

1. keep the add_argument for 'ros2_distro' for just in case in the future usage.
2. remove the add_argument since there is no usage in current version.

Can you make any suggestion for this?

Reply:
@ldg810 Currently `jinga_gen.py` is run automatically at build time in https://github.com/PX4/PX4-SITL_gazebo/blob/4f26815d4127e3af84e01bbc1f8683c4fe30521f/CMakeLists.txt#L275 . My preference would be that we detect whether the environment is for ros2 or ros1 as done in https://github.com/PX4/PX4-SITL_gazebo/blob/4f26815d4127e3af84e01bbc1f8683c4fe30521f/CMakeLists.txt#L116 and pass the `ros2_distro` argument from the cmake script. Does this make sense?

I am a bit unclear how `generate_microRTPS_bridge.py` fits in the context of this PR

Reply:
@ldg810 I like the suggestion from @Jaeyoung-Lim. Can you please replace it? Then this should be good to go.